code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -0,0 +1,92 @@
+//Engineering Mesons
+
+/obj/item/clothing/glasses/meson/engine
+	name = "Engineering Scanner Goggles"
+	desc = "Goggles used by engineers. The Meson Scanner mode lets you see basic structural and terrain layouts through walls, regardless of lighting condition. The T-ray Scanner mode lets you see underfloor objects such as cables and pipes."
+	icon_state = "trayson-meson"
+	item_state = "glasses"
+	origin_tech = "magnets=2;engineering=2"
+	darkness_view = 1
+	vision_flags = SEE_TURFS
+	invis_view = SEE_INVISIBLE_MINIMUM
```

Comment:
invis_view, vision_flags, darkness_view and origin_tech are already defined in /obj/item/clothing/glasses/meson . So no need to add them here again